internal/node/comments.go

```diff
@@ -0,0 +1,105 @@
+package node
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+// ---------------------------------------------------------------------------
+// comments
+// ---------------------------------------------------------------------------
+
+type Comment struct {
+	// the func name in which the comment was created.
+	Caller string
+	// the name of the file owning the caller.
+	File string
+	// the comment message itself.
+	Message string
+}
+
+// shorthand for checking if an empty comment was generated.
+func (c Comment) IsEmpty() bool {
+	return len(c.Message) == 0
+}
+
+// NewComment formats the provided values, and grabs the caller and trace
+// info according to the depth.  Depth is a skip-caller count, and any func
+// calling this one should provide either `1` (for itself) or `depth+1` (if
+// it was already given a depth value).
+func NewComment(
+	depth int,
+	template string,
+	values ...any,
+) Comment {
+	caller := GetCaller(depth + 1)
+	_, _, parentFileLine := GetDirAndFile(depth + 1)
+
+	return Comment{
+		Caller:  caller,
+		File:    parentFileLine,
+		Message: fmt.Sprintf(template, values...),
+	}
+}
+
+// AddComment creates a new nodewith a comment but no other properties.
+func (dn *Node) AddComment(
```

Comment:
I see that `TestAddComment` exists but in `clues_test.go`. Could you move it to `comments_test.go` file in a future PR?

Reply:
can do!